src/main.py

```diff
@@ -18,28 +20,97 @@ async def on_ready():
-    other = reaction.message.mentions[0]
-    embed = discord.Embed(title=f"Some amazing art by {str(other)}", type="rich", color=other.color, description=txt[-2])
-    embed.set_image(url=txt[-1])
-    gallery = client.get_channel(GALLERY_CHAN)
-    await gallery.send(embed=embed)
+
+    # Staff reactions to verification channel posts by this client
+    if reaction.message.channel.id == VERIFY_CHAN:
+        # Ignore reactions to posts not by this client in verification channel
+        if client.user != reaction.message.author:
+            return
+
+        # Publish verified posts to gallery channel
+        await publish_art(message=reaction.message)
+
+    # User reactions to posts in self-curated channel
+    elif reaction.message.channel.id in MOD_CHANS:
+        # Ignore reactions from users without any of the required roles
+        if len(SHOWCASE_ROLES) == 0 or len([r for r in user.roles if r.id in SHOWCASE_ROLES]) == 0:
+            return
+
+        # Ignore reactions from users other than the message author
+        if user != reaction.message.author:
+            return
+
+        # Publish self-curated posts
+        await reaction.message.add_reaction(emoji=reaction.emoji)
+        await publish_mod(message=reaction.message, reaction=reaction)
+
+    # User reactions to posts in showcase channel
+    elif reaction.message.channel.id == SHOWCASE_CHAN:
+        # Ignore reactions from users other than the message author
+        if user != reaction.message.author:
+            return
+
+        # Remove bulletin posts
+        await reaction.message.delete()
 
 @client.event
-async def on_message(message):
+async def on_message(message: discord.Message):
     # Ignore all bots
     if message.author.bot:
         return
 
+    # Send posts from creative channel to verification channel
     if message.channel.id in ART_CHANS:
-        verify = client.get_channel(VERIFY_CHAN)
-        for img in message.attachments:
-            await verify.send(f"<@{message.author.id}> has posted:\n{message.content}\n{img.url}")
+        await verify_art(message=message)
+
+async def verify_art(message: discord.Message):
+    verify = client.get_channel(id=VERIFY_CHAN)
+    for img in message.attachments:
+        await verify.send(f"<@{message.author.id}> has posted:\n{message.content}\n{img.url}")
+
+async def publish_art(message: discord.Message):
+    gallery = client.get_channel(id=GALLERY_CHAN)
+    user = message.mentions[0]
+    title = f"Some amazing art by {user}"
+    split = message.split('\n')
+    text = split[1:-2]
+    embed = discord.Embed(title=title, type="rich", color=user.color, description=text)
+    embed.set_image(url=split[-1])
+    await send_embed(channel=gallery, embed=embed, message=message)
+
+async def publish_mod(message: discord.Message, reaction: discord.Reaction):
+    channel = client.get_channel(id=SHOWCASE_CHAN)
+    user = message.author
+    title = f"{reaction.emoji}   posted in #{str(message.channel)}"
```

Comment:
Note: extra spaces here. I'm fine if that's intentional

Reply:
Intentional, but Discord seems to flatten grouped spaces, so it has no effect. Can be replaced with a single space.